### PR TITLE
CUMULUS-455: update lambda zip file when message adapter is updated

### PR DESCRIPTION
**Pull request by jennyhliu**
**Summary:** Summary of changes

Addresses [CUMULUS-455: Kes deployments using only an updated message adapter do not get automatically deployed(https://bugs.earthdata.nasa.gov/browse/CUMULUS-455)

## Changes

*  prepended the hash value of cumulus-message-adapter.zip file to the zip file name of lambda which uses message adapter.
* the lambda function will be redeployed when message adapter or lambda function are updated

## Test Plan
Things that should succeed before merging.

- [x ] Unit tests
- [x ] Adhoc testing: deployment with the updated kes
- [x ] Update CHANGELOG
- [x] Run `./bin/eslint-ratchet` and verify that eslint errors have not increased
  - [x] Commit `.eslint-ratchet-high-water-mark` if the score has improved

## Release PR

If this is a **release** PR, make sure you branch name starts with `release-` prefix, e.g. `release-v-1.1.2`.

Reviewers: @tag-your-friends

### CODE REVIEW

File: packages/deployment/lib/kes.js [hunk 1]
```diff
@@ -151,14 +152,16 @@ class UpdatedKes extends Kes {
     const src = path.join(process.cwd(), kesBuildFolder, filename);
     const dest = path.join(process.cwd(), kesBuildFolder, 'adapter', unzipFolderName);
 
-    // return Promise.resolve();
+    if (!filename) return Promise.resolve();
```
Comment (laurenfrederick) [inline packages/deployment/lib/kes.js]: So this would not call `super` if there is no message adapter filename? That doesn't seem right to me.

Comment (jennyhliu) [reply packages/deployment/lib/kes.js]: Ah, wanted to simplify the code.  Updated.

File: packages/deployment/lib/kes.js [hunk 2]
```diff
@@ -151,14 +152,16 @@ class UpdatedKes extends Kes {
...
+    if (!filename) return super.compileCF();
     return fetchMessageAdapter(
       this.config.message_adapter_version,
       this.messageAdapterGitPath,
       filename,
       src,
       dest
-    ).then(() => super.compileCF());
+    ).then(() => {
+      this.Lambda.messageAdapterZipFileHash = new this.Lambda(this.config).getHash(src);
```
Comment (scisco) [inline packages/deployment/lib/kes.js]: @jennyhliu the problem with this approach is that the hash only changes if the message adapter is changed. this means that if you make a change to your lambda function, it never gets updated.

Comment (jennyhliu) [reply packages/deployment/lib/kes.js]: The hash of the message adapter is prepend to the lambda zip file name, so one of the hash values is changed, the lambda zip file name is changed, the zipLambda will create a new zip file. e.g.
Zipping app/build/cloudformation/da2198a240ed8948234f5e462236cbc57a6fed49-2354f9a4959ba95c1e7c8d12a09f5c09e573749a-ParsePdr.zip and injecting message adapter for ParsePdr
When I updated the existing zip file name.

Comment (scisco) [reply packages/deployment/lib/kes.js]: oh, right. so the lambda name already has the hash in it. didn't realize that. should be worried about the filename length?

Comment (jennyhliu) [reply packages/deployment/lib/kes.js]: S3 Max Key Length is 1024, and mac file name limit is 255.  We should be fine.  We can change to use cksum if it becomes an issue in the future.

**Pull request merged**